docs/src/main/paradox/google-cloud-bigquery.md

```diff
@@ -94,10 +95,30 @@ You can use the built-in @apidoc[BigQueryCallbacks$].
 
 ### Parsers
 
-The parser function is a @scala[`spray.json.JsObject => Try[T]`]@java[`java.util.function.Function[spray.json.JsObject, scala.util.Try[T]]`] function. 
-This is needed because there is a possibility, the response not to contain any data. In this case we need to retry the request with some delay.
-Your parser function needs to be bulletproof, and the code in the examples represents the happy path.
-In case of `scala.util.Failure` your stream will be polling forever!
+The provided parser is implemented with the Spray JSON library.
+@scala[To use, bring into scope the required implicits with `import akka.stream.alpakka.googlecloud.bigquery.scaladsl.SprayJsonSupport._` and provide either an implicit `spray.json.JsonFormat[T]` for `GoogleBigQuerySource[T].runQuery` or `spray.json.RootJsonFormat[T]` for `GoogleBigQuerySource[T].raw`.]
+@java[To use, you must provide a `java.util.function.Function<spray.json.JsObject, scala.util.Try<T>>` function.]
+
+When calling the `GoogleBigQuerySource.raw` method, your parser is passed the [entire response body](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/getQueryResults#response-body).
+When calling the `GoogleBigQuerySource.runQuery` method, your parser is passed one-by-one each entry in the `rows` key of the response.
+Note that these entries are represented as "a series of JSON f,v objects for indicating fields and values" ([reference documentation](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/getQueryResults#body.GetQueryResultsResponse.FIELDS.rows)).
+@scala[To correctly generate a Spray `JsonFormat[T]` for your case class `T`, you must `import akka.stream.alpakka.googlecloud.bigquery.scaladsl.BigQueryJsonProtocol._` and use the provided `bigQueryJsonFormatN(T)` methods as a drop-in replacement for the usual `jsonFormatN(T)`. Furthermore, the order and presence of parameters in `T` is required to strictly match the query.]
```

Comment:
I'm not sure, that BQ can't mess up the param order. I hope they didn't go that far, but would be good to check it...

Reply:
You raise an important point; I'll look into it.

Reply:
[This stackoverflow answer](https://stackoverflow.com/a/59825040) suggests this it is "pretty safe" to assume this.

Reply:
I hope so. My only concern is that there is a "schema" field that returns the explicit return order. I hope that they don't mess up things for sport :D